include/mxnet/ndarray.h

```diff
@@ -783,6 +792,11 @@ class NDArray {
     /*! This is created when data is stored in MKLDNN format.
      */
     std::shared_ptr<MKLDNNMemory> mkl_mem_;
+#endif
+#if MXNET_USE_NGRAPH == 1
+    /*! this is set if ngraph tensorview is associated with this ndarray
+     */
+    std::shared_ptr<ngraph::runtime::TensorView> tensor_view_;
```

Comment:
does ngraph has to add a field in NDArray?

Reply:
This is needed to get performance on backends other than GPU in Imperative Gluon (think MKLDNN layouts or non CPU/non GPU memory). We're not including these other backends in this first PR, and we haven't fully optimized Gluon for CPU through nGraph, but this is setting the stage.

Reply:
it's kind of weird to me that the nGraph code relies on MXNet.

Reply:
for performance reasons, nGraph mxnet integration needs to reuse memory allocated by mxnet (avoid reallocation/memcpy etc), and also uniquely map ndarray with ngraph integration.

Reply:
@zheng-da nGraph proper doesn't depend on MXNet, but the glue code that translates between IRs and provides an execution interface to MXNet does. We separated that code into `3rdparty/ngraph-mxnet-bridge` to reduce the burden on MXNet's build system and reduce the changes to MXNet's other execution paths.